core/iwasm/common/wasm_c_api.c

```diff
@@ -1924,7 +1924,8 @@ wasm_frame_func_offset(const wasm_frame_t *frame)
 void
 wasm_frame_vec_clone_internal(Vector *src, Vector *out)
 {
-    bh_assert(src->num_elems != 0 && src->data);
+    if (src->num_elems == 0)
+        return;
```

Comment:
Had better to `bh_vector_destory(out)` before return to free its possible frames and zero-initialize out? For example, an instance's frames may be created previously and then `wasi_proc_exit` occurs.

BTW, in L1932, seems no need to destroy out again if bh_vector_init fails?